Purple.Vipers.Api/Controllers/CatalogController.cs

```diff
@@ -18,44 +20,74 @@ public CatalogController(StoreContext db)
         [HttpGet]
         public IActionResult GetItems()
         {
+
             return Ok(_db.Items);
         }
 
         [HttpGet("{id:int}")]
         public IActionResult GetItem(int id)
         {
-            var item = new Item("Shirt", "Ohio State shirt.", "Nike", 29.99m);
-            item.Id = id;
-            
-            return Ok(item);
+            var item = _db.Items.Find(id);
+            if (item == null)
+            {
+                return NotFound();
+            }
+            return Ok();
         }
 
         [HttpPost]
         public IActionResult Post(Item item)
         {
-            return Created("/catalog/42", item);
+            _db.Items.Add(item);
+            _db.SaveChanges();
+            return Created($"/catalog/{item.Id}", item);
         }
 
         [HttpPost("{id:int}/ratings")]
         public IActionResult PostRating(int id, [FromBody] Rating rating)
         {
-            var item = new Item("Shirt", "Ohio State shirt.", "Nike", 29.99m);
-            item.Id = id;
+            var item = _db.Items.Find(id);
+            if (item == null) {
+                return NotFound();
+            }
             item.AddRating(rating);
+            _db.SaveChanges();
 
             return Ok(item);
         }
 
         [HttpPut("{id:int}")]
         public IActionResult Put(int id, Item item)
```

Comment:
Need to make some minor adjustments. Should be 
public IActionResult Put**Item**(int id, **[FromBody]** Item item)